### ISSUE TITLE
Global variables are not connected to the self and window objects

### ISSUE THREAD

**Issue by bourey** [ampproject/worker-dom#177]
When running code with WorkerDOM, it appears that global variables don't show up on the self and window objects, and variables set on self or window can't be referenced as globals. This causes errors for code using Google Closure or ZoneJS libraries (and Angular, given its dependency on Zone).

For a small repro, try loading the following code via the main thread, then via a WorkerDOM-powered web worker:
```
console.log('Object is defined on self', !!self.Object);
console.log('Object is defined on window', !!window.Object);

window.Cat = 'Meow';
console.log('Set global variable Cat on the window', Cat);

self.Dog = 'Woof';
console.log('Set global variable Foo on the window', Dog);
```

Outside of a Web Worker, the script prints out the following console output:
```
Object is defined on self true
Object is defined on window true
Set global variable Cat on the window Meow
Set global variable Foo on the window Woof
```

When loaded with WorkerDOM, the code outputs:
```
Object is defined on self false
Object is defined on window false
Uncaught ReferenceError: Cat is not defined
```

**Comment by kristoferbaxter** [comment]
@bourey – Just pushed a branch designed to [resolve this issue](https://github.com/ampproject/worker-dom/tree/with-window). 

Would you mind trying it out for your needs and reporting back? If it works, we'll add another output target for the worker code to support when a Babel transform cannot be run on the worker code to solve this problem.

**Comment by kristoferbaxter** [comment]
Makes sense, going to take a different approach to try and solve this (augment existing globals in the unsafe mode versus overwrite or wrap author code in a custom closure). Likely the next update will be in a week.

**Comment by kristoferbaxter** [comment]
Yep! We can do this augmentation during worker creation so only worker code is impacted.

**Comment by choumx** [comment]
This was fixed in #578.
